controllers/istio_controller.go

```diff
@@ -171,15 +156,38 @@ func (r *IstioReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	return ctrl.Result{}, err
 }
 
-func applyOverrides(istio *v1alpha1.Istio, values helm.HelmValues) (helm.HelmValues, error) {
-	if values == nil {
-		values = helm.HelmValues{}
+func (r *IstioReconciler) getAggregatedValues(istio v1alpha1.Istio) (helm.HelmValues, error) {
+	// 1. start with values aggregated from default profiles and the profile in Istio.spec.profile
+	values, err := getValuesFromProfiles(r.getProfilesDir(istio), r.getProfiles(istio))
+	if err != nil {
+		return nil, err
 	}
 
+	// 2. apply values from Istio.spec.values, overwriting values from profiles
+	values = mergeOverwrite(values, istio.Spec.GetValues())
+
+	// 3. apply values from Istio.spec.rawValues, overwriting the current values
```

Comment:
so, rawValues has precedence over values... We should document this somewhere...

Reply:
BTW, do we have tests covering this?

Reply:
@luksa  Should it be `values` that has precedence over `rawValues`?

Reply:
Yes. in @bmangoen's initial iplementation, values had precedence over rawValues, but I've now purposefully changed this, because this is also the case in the IstioOperator resource (`unvalidatedValues` overrides `values`).